cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/CouchDbException.java

```diff
@@ -81,10 +76,11 @@ public String getReason() {
     @Override
     public String getMessage() {
         //improve the message with status code, error and reason if we have them
-        return ((getStatusCode() > 0) ? getStatusCode() + " " : "") + super.getMessage()
-                + ((url != null) ? ": " + url : "")
-                + ((error != null) ? ": " + getError() : "")
-                + ((reason != null) ? ": " + getReason() : "");
+        String msg = super.getMessage().replaceAll("\\.$", "");
```

Comment:
Sure, it does seem overkill to compile this same regex each time. I wonder how a pre-compiled pattern would fare. Either way, I'll update with your suggestion. Thanks.